eppo_core/src/ufc/assignment.rs

```diff
@@ -0,0 +1,251 @@
+    /// let value = AssignmentValue::Boolean(true);
+    /// assert_eq!(value.as_boolean(), Some(true));
+    /// ```
+    pub fn as_boolean(&self) -> Option<bool> {
+        match self {
+            AssignmentValue::Boolean(b) => Some(*b),
+            _ => None,
+        }
+    }
+
+    /// Checks if the assignment value is of type Json.
+    ///
+    /// # Returns
+    /// - `true` if the value is of type Json, otherwise `false`.
+    ///
+    /// # Examples
+    /// ```
+    /// # use eppo_core::ufc::AssignmentValue;
+    /// use serde_json::json;
+    ///
+    /// let value = AssignmentValue::Json(json!({ "key": "value" }));
+    /// assert_eq!(value.is_json(), true);
+    /// ```
+    pub fn is_json(&self) -> bool {
+        self.as_json().is_some()
+    }
+    /// Returns the assignment value as a JSON value if it is of type Json.
+    ///
+    /// # Returns
+    /// - The JSON value if the assignment value is of type Json, otherwise `None`.
+    ///
+    /// # Examples
+    /// ```
+    /// # use eppo_core::ufc::AssignmentValue;
+    /// use serde_json::json;
+    ///
+    /// let value = AssignmentValue::Json(json!({ "key": "value" }));
+    /// assert_eq!(value.as_json(), Some(&json!({ "key": "value" })));
+    /// ```
+    pub fn as_json(&self) -> Option<&serde_json::Value> {
+        match self {
+            Self::Json(v) => Some(v),
+            _ => None,
+        }
+    }
+    /// Extracts the assignment value as a JSON value if it is of type Json.
+    ///
+    /// # Returns
+    /// - The JSON value if the assignment value is of type Json, otherwise `None`.
+    ///
+    /// # Examples
+    /// ```
+    /// # use eppo_core::ufc::AssignmentValue;
+    /// use serde_json::json;
+    ///
+    /// let value = AssignmentValue::Json(json!({ "key": "value" }));
+    /// assert_eq!(value.to_json(), Some(json!({ "key": "value" })));
+    /// ```
+    pub fn to_json(self) -> Option<serde_json::Value> {
+        match self {
+            Self::Json(v) => Some(v),
+            _ => None,
+        }
+    }
+}
+
+/// Represents an event capturing the assignment of a feature flag to a subject and its logging
+/// details.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AssignmentEvent {
+    /// The key of the feature flag being assigned.
+    pub feature_flag: String,
+    /// The key of the allocation that the subject was assigned to.
+    pub allocation: String,
+    /// The key of the experiment associated with the assignment.
+    pub experiment: String,
+    /// The specific variation assigned to the subject.
+    pub variation: String,
+    /// The key identifying the subject receiving the assignment.
+    pub subject: String,
+    /// Custom attributes of the subject relevant to the assignment.
+    pub subject_attributes: Attributes,
+    /// The timestamp indicating when the assignment event occurred.
+    pub timestamp: String,
```

Comment:
I'm leading a low key crusade to having the timestamp in our assignment events be language-native dates

Reply:
I generally agree but it might be harder to get native date type across the language borders. String is a safe bet.